backend/src/services/rate-limit/rate-limit-service.ts

```diff
@@ -0,0 +1,68 @@
+import { CronJob } from "cron";
+
+import { getConfig } from "@app/lib/config/env";
+import { ForbiddenRequestError } from "@app/lib/errors";
+import { logger } from "@app/lib/logger";
+import { rateLimitMaxConfiguration } from "@app/server/config/rateLimiter";
+
+import { TRateLimitDALFactory } from "./rate-limit-dal";
+import { TRateLimit, TRateLimitUpdateDTO } from "./rate-limit-types";
+
+type TRateLimitServiceFactoryDep = {
+  rateLimitDAL: TRateLimitDALFactory;
+};
+
+export type TRateLimitServiceFactory = ReturnType<typeof rateLimitServiceFactory>;
+
+export const rateLimitServiceFactory = ({ rateLimitDAL }: TRateLimitServiceFactoryDep) => {
+  const getRateLimits = async (): Promise<TRateLimit | undefined> => {
+    try {
+      return await rateLimitDAL.findOne({ id: "00000000-0000-0000-0000-000000000000" });
+    } catch (error) {
+      return undefined;
+    }
+  };
+
+  const updateRateLimit = async (updates: TRateLimitUpdateDTO): Promise<TRateLimit> => {
+    const appCfg = getConfig();
+    if (!appCfg.ALLOW_RATELIMIT_UPDATES) {
+      throw new ForbiddenRequestError({
+        name: "Rate limit Updates Disabled",
+        message: "Changes to rate limits are disabled"
+      });
+    }
```

Comment:
for infisical cloud

Reply:
I don't think for infisical cloud we need to disable this. I mean its super admin only right.

CC: @maidul98

Reply:
ah, was confused about who the super admin is. will address